src/measure_theory/measure/vector_measure.lean

```diff
@@ -545,6 +545,24 @@ begin
   { exact dif_neg hf }
 end
 
+/-- Given a vector measure `v` on `M` and a continuous add_monoid_hom `f : M → N`, `f ∘ v` is a
+vector measure on `N`. -/
+def map_range {N : Type*} [add_comm_monoid N] [topological_space N]
+  (v : vector_measure α M) (f : M →+ N) (hf : continuous f) : vector_measure α N :=
```

Comment:
Can you define this as an `add_monoid_hom` from `vector_measure α M` to `vector_measure α N`? That way your `re` and `im` would also be `add_monoid_hom` and you get `re_add` and `re_zero` for free.

Reply:
To defined a `add_monoid_hom` on vector measures we will need `has_continuous_add` on both `M` and `N`. I will make a new definition for this version

Reply:
I've added the linear map version as well for good measure.